src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloudSnapshotCreatedIT.java

```diff
@@ -0,0 +1,147 @@
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class ComputeEngineCloudSnapshotCreatedIT {
+  private static Logger log = Logger.getLogger(ComputeEngineCloudSnapshotCreatedIT.class.getName());
+
+  private static final String SNAPSHOT_LABEL = "snapshot";
+  private static final int SNAPSHOT_TEST_TIMEOUT = 120;
+
+  @ClassRule public static Timeout timeout = new Timeout(5, TimeUnit.MINUTES);
+  @ClassRule public static JenkinsRule jenkinsRule = new JenkinsRule();
+
+  private static ByteArrayOutputStream logOutput = new ByteArrayOutputStream();
+  private static StreamHandler streamHandler = new StreamHandler(logOutput, new SimpleFormatter());
+  private static ComputeClient client;
+  private static Map<String, String> label = getLabel(ComputeEngineCloudSnapshotCreatedIT.class);
+  private static Snapshot createdSnapshot = null;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    log.info("init");
+    initCredentials(jenkinsRule);
+    ComputeEngineCloud cloud = initCloud(jenkinsRule);
+    handleClassLogs(streamHandler, ComputeEngineCloud.class.getName());
+    client = initClient(jenkinsRule, label, log);
+    handleClassLogs(streamHandler, ComputeClient.class.getName());
+
+    assertTrue(cloud.configurations.isEmpty());
+    InstanceConfiguration instanceConfiguration =
+        instanceConfiguration(
+            DEB_JAVA_STARTUP_SCRIPT,
+            NUM_EXECUTORS,
+            SNAPSHOT_LABEL,
+            label,
+            true,
+            true,
+            NULL_TEMPLATE);
+    cloud.addConfiguration(instanceConfiguration);
+    assertTrue(
+        logs(streamHandler, logOutput),
+        cloud.getInstanceConfig(instanceConfiguration.description).createSnapshot);
+
+    // Assert that there is 0 nodes
+    assertTrue(jenkinsRule.jenkins.getNodes().isEmpty());
+
+    FreeStyleProject project = jenkinsRule.createFreeStyleProject();
+    Builder step = new Shell("exit 1");
+    project.getBuildersList().add(step);
+    project.setAssignedLabel(new LabelAtom(SNAPSHOT_LABEL));
+
+    FreeStyleBuild build = jenkinsRule.assertBuildStatus(Result.FAILURE, project.scheduleBuild2(0));
+    Node worker = build.getBuiltOn();
+    assertNotNull(logs(streamHandler, logOutput), worker);
+    // Cannot handle class logs for ComputeEngineInstance until an instance exists.
+    handleClassLogs(streamHandler, ComputeEngineInstance.class.getName());
+
+    // Need time for one-shot instance to terminate and create the snapshot
+    Awaitility.await()
+        .timeout(SNAPSHOT_TEST_TIMEOUT, TimeUnit.SECONDS)
+        .until(() -> jenkinsRule.jenkins.getNode(worker.getNodeName()) == null);
+
+    createdSnapshot = client.getSnapshot(PROJECT_ID, worker.getNodeName());
+  }
+
+  @AfterClass
+  public static void teardown() throws IOException {
+    if (createdSnapshot != null) {
+      client.deleteSnapshot(PROJECT_ID, createdSnapshot.getName());
+    }
```

Comment:
Add '\n' after closing brace.

Reply:
After re-reading the Google java style guidelines I realized I was conflating some aspects with Go's style guidelines. In particular the '\n' after closing branch/loop block braces. Disregard.